server/kapacitors.go

```diff
@@ -299,10 +313,46 @@ func (s *Service) UpdateKapacitor(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if activateKapacitor {
+		// make sure that there is at most one active kapacitor
+		err := s.deactivateOtherKapacitors(ctx, srcID, id)
+		if err != nil {
+			Error(w, http.StatusInternalServerError, err.Error(), s.Logger)
+			return
+		}
+	}
+
 	res := newKapacitor(srv)
 	encodeJSON(w, http.StatusOK, res, s.Logger)
 }
 
+// deactivateOtherKapacitors deactivates all kapacitors excluding the one with supplied ID
+func (s *Service) deactivateOtherKapacitors(ctx context.Context, srcID int, ID int) error {
+	serversStore := s.Store.Servers(ctx)
+	mrSrvs, err := serversStore.All(ctx)
+	if err != nil {
+		return errors.New("error loading kapacitors for deactivation")
+	}
+	var deactivationErrors []string = nil
+	var deactivationError error = nil
+	for _, srv := range mrSrvs {
+		if srv.SrcID == srcID && srv.Type == "" && srv.ID != ID {
+			if srv.Active {
+				srv.Active = false
+				if err := serversStore.Update(ctx, srv); err != nil {
+					deactivationErrors = append(deactivationErrors, err.Error())
+					deactivationError = err
```

Comment:
this isn't required as strings.Join works with a value of a single length. https://play.golang.org/p/qCHrfQP1Tgd

Reply:
@glinton ...it is required to return the original error for the existing tests to pass. I wanted to use [error wrapping](https://blog.golang.org/go1.13-errors) to have simpler code, but the tests failed to compile in CI job, they are probably run on pre 1.13 go. I don't want to dig deeper here, thank you for your review.